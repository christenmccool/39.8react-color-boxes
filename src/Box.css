.Box {
  display: flex;
  align-items: flex-start;
  gap: 5px;
}

.Box-btn {
  /* height: max-content; */
  /* align-self: flex-start; */
}