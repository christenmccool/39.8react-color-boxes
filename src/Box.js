import React from 'react';
import './Box.css'

const Box = ( {id, color, width, height, removeBox}) => {
  const style = {
    backgroundColor: color,
    width: parseInt(width),
    height: parseInt(height)
  }

  return (
    <div className="Box">
      <div className="Box-box" style={style}></div>
      <button className="Box-btn" onClick={removeBox}>X</button>
    </div>
  )
}

export default Box;