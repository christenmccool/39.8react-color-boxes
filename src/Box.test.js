import { render, screen } from '@testing-library/react';
import Box from './Box';

test('if renders without crashing', () => {
  render(<Box />);
});

test("if matches snapshot", () => {
  const {asFragment} = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
})

