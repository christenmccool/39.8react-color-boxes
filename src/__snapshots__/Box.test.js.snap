// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <div
      class="Box-box"
    />
    <button
      class="Box-btn"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
